File: Parallel-Csharp/src/Parallel/Base/Global.cs
using Parallel.Entity;

namespace Parallel.Base;

public class Global
{
    private EntityCollection _globalEntityCollection;

    public EntityCollection GlobalEntityCollection => _globalEntityCollection;

    private static Global? _instance;

    private Global()
    {
        _globalEntityCollection = new EntityCollection();
    }

    public static int GenerateEntityId()
    {
        return 1;
    }
    public static Global Instance
    {
        get
        {
            if (_instance == null)
                _instance = new Global();
            return _instance;
        }
    }
}

File: Parallel-Java/src/parallel/base/Global.java
package parallel.base;

import parallel.entity.EntityCollection;

public class Global {
    private EntityCollection globalEntityCollection;

    public EntityCollection getGlobalEntityCollection() {
        return globalEntityCollection;
    }
    private static Global instance;

    private Global() {
        globalEntityCollection = new EntityCollection();
    }

    public static int GenerateEntityId() {
        return 1;
    }

    public static Global getInstance() {
        if (instance == null)
            instance = new Global();
        return instance;
    }
}